### ISSUE TITLE
Blocking feature does not work

### ISSUE THREAD

**Issue by gsmetal** [deanpcmad/sidekiq-limit_fetch#71]
I have following in my sidekiq.yml:

``` yaml
:queues:
  - create
  - delete
  - default
:blocking:
  - create
```

Then I start sidekiq and try adding job for `default` queue using `TestJob.perform_later`. Nothing happens. If I add job for `create` queue (by `TestJob.set(queue: "create").perform_later`), it's processed. Only when I manually do `Sidekiq::Queue['create'].unblock` `default` queue jobs starts processing.

I am using `Sidekiq 4.1.1` and `Redis server v=3.0.3 sha=00000000:0 malloc=jemalloc-3.6.0 bits=64 build=58013d157c63182b`

**Comment by ivanovaleksey** [comment]
The same issue for me. 
Any jobs in queues with lower priority than blocking queue are not being executed at all.

**Comment by ivanovaleksey** [comment]
It seems to be working with previous versions.
I tried to use `sidekiq-3.5.4` and `sidekiq-limit_fetch-2.4.2`. The blocking feature works fine but sidekiq gives me errors on start. Here is backtrace
```
Thread.exclusive is deprecated, use Mutex
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/sidekiq-limit_fetch-2.4.2/lib/sidekiq/limit_fetch/global/selector.rb:18:in `uuid'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/sidekiq-limit_fetch-2.4.2/lib/sidekiq/limit_fetch/global/monitor.rb:65:in `heartbeat_key'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/sidekiq-limit_fetch-2.4.2/lib/sidekiq/limit_fetch/global/monitor.rb:47:in `block (2 levels) in update_heartbeat'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/redis-3.3.2/lib/redis.rb:2303:in `block in multi'
Thread.exclusive is deprecated, use Mutex/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/redis-3.3.2/lib/redis.rb:58:in `block in synchronize'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/sidekiq-limit_fetch-2.4.2/lib/sidekiq/limit_fetch/global/selector.rb:18:in `uuid'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/sidekiq-limit_fetch-2.4.2/lib/sidekiq/limit_fetch/global/selector.rb:6:in `acquire'

/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/sidekiq-limit_fetch-2.4.2/lib/sidekiq/limit_fetch/queues.rb:18:in `acquire'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/sidekiq-limit_fetch-2.4.2/lib/sidekiq/limit_fetch.rb:37:in `fetch_message'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/redis-3.3.2/lib/redis.rb:58:in `synchronize'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/redis-3.3.2/lib/redis.rb:2296:in `multi'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/sidekiq-limit_fetch-2.4.2/lib/sidekiq/limit_fetch.rb:30:in `retrieve_work'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/sidekiq-3.5.4/lib/sidekiq/fetch.rb:37:in `block in fetch'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/sidekiq-limit_fetch-2.4.2/lib/sidekiq/limit_fetch/global/monitor.rb:46:in `block in update_heartbeat'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/sidekiq-3.5.4/lib/sidekiq.rb:84:in `block in redis'

/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/connection_pool-2.2.1/lib/connection_pool.rb:64:in `block (2 levels) in with'/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/sidekiq-3.5.4/lib/sidekiq/util.rb:16:in `watchdog'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/sidekiq-3.5.4/lib/sidekiq/fetch.rb:33:in `fetch'

/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/calls.rb:28:in `public_send'/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/connection_pool-2.2.1/lib/connection_pool.rb:63:in `handle_interrupt'

/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/calls.rb:28:in `dispatch'/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/connection_pool-2.2.1/lib/connection_pool.rb:63:in `block in with'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/connection_pool-2.2.1/lib/connection_pool.rb:60:in `handle_interrupt'

/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/connection_pool-2.2.1/lib/connection_pool.rb:60:in `with'/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/call/async.rb:7:in `dispatch'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/cell.rb:50:in `block in dispatch'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/sidekiq-3.5.4/lib/sidekiq.rb:81:in `redis'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/sidekiq-limit_fetch-2.4.2/lib/sidekiq/limit_fetch/global/monitor.rb:45:in `update_heartbeat'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/cell.rb:76:in `block in task'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/sidekiq-limit_fetch-2.4.2/lib/sidekiq/limit_fetch/global/monitor.rb:15:in `block (2 levels) in start!'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/actor.rb:339:in `block in task'

/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/sidekiq-limit_fetch-2.4.2/lib/sidekiq/limit_fetch/global/monitor.rb:13:in `loop'/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/task.rb:44:in `block in initialize'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/celluloid-0.17.3/lib/celluloid/task/fibered.rb:14:in `block in create'
/Users/aleksey/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/sidekiq-limit_fetch-2.4.2/lib/sidekiq/limit_fetch/global/monitor.rb:13:in `block in start!'
```

**Comment by sharvy** [comment]
By the way, Sidekiq already provides basic blocking facility. According to their [page](https://github.com/mperham/sidekiq/wiki/Advanced-Options):

> If you want queues always processed in a specific order, just declare them in order without weights:
```
:queues:
  - critical
  - default
  - low
```
> This means that any job in the default queue will be processed only when the critical queue is empty.

**Comment by abhi-sm** [comment]
@sharvy well blocking does more than setting the priority. The doc says:

>  if a blocking queue task is executing then no new task from lesser priority queues will be ran

Which I think means no other task will be taken up by other queues when a blocking queue task is in processing state.

**Comment by abhiQmar** [comment]
@sharvy well blocking does more than setting the priority. The doc says:

> if a blocking queue task is executing then no new task from lesser priority queues will be ran

Which I think means no other task will be taken up by other queues when a blocking queue task is in processing state.

**Comment by brandoncc** [comment]
@brainopia I'm not familiar with redis internals, can you give me a hint where you think the problem is so I can work on fixing this? This gem is awesome but this broken feature is causing us a lot of problems.

**Comment by brandoncc** [comment]
Actually, I think not being familiar with sidekiq internals is more detrimental to me fixing it.

**Comment by deanpcmad** [comment]
Hey, thanks for your issue! I've just taken over this project/gem and am going through the PRs and issues. I see this issue is stale so I'm going to close it but if you are still experiencing issues, please feel free to re-open.
